recipe/meta.yaml

```diff
@@ -13,58 +13,83 @@ source:
+  missing_dso_whitelist:         # [linux and (x86_64 or s390x)]
+    - '**/libc.so.6'             # [linux and x86_64]
+    - '**/libpthread.so.0'       # [linux and x86_64]
+    - '**/libdl.so.2'            # [linux and x86_64]
+    - '**/librt.so.1'            # [linux and x86_64]
+    - '**/libm.so.6'             # [linux and x86_64]
+    - '**/ld-linux-x86-64.so.2'  # [linux and x86_64]
+    - '**/ld64.so.1'             # [linux and s390x]
+  ignore_run_exports:
+    - pixman
 
 requirements:
   build:
     - cmake  # [win]
     - make  # [unix]
-    - gobject-introspection
+    - gobject-introspection # [not win]
     - pkg-config
     - {{ compiler('rust') }}
     - {{ compiler('c') }}
-    - gnuconfig  # [unix]
-    - glib        # [build_platform != target_platform]
-    - libxml2     # [build_platform != target_platform]
-    - cairo       # [build_platform != target_platform]
-    - gdk-pixbuf  # [build_platform != target_platform]
-    - pango       # [build_platform != target_platform]
-    - {{ cdt('libxau-devel') }}          # [linux and aarch64]
-    - {{ cdt('xorg-x11-util-macros') }}  # [linux and aarch64]
-    - {{ cdt('xorg-x11-proto-devel') }}  # [linux and aarch64]
-    - {{ cdt('libx11-devel') }}          # [linux and aarch64]
-    - {{ cdt('libx11-common') }}         # [linux and aarch64]
-    - {{ cdt('libx11') }}                # [linux and aarch64]
-    - {{ cdt('libxext-devel') }}         # [linux and aarch64]
-    - {{ cdt('libxext') }}               # [linux and aarch64]
+    - autoconf    # [unix]
+    - automake    # [unix]
+    - libtool     # [unix]
+    - patch       # [unix]
+    - m2-patch    # [win]
+    - gnuconfig   # [unix]
+    - pthread-stubs                      # [linux]
+    - {{ cdt('libxau-devel') }}          # [linux]
+    - {{ cdt('libxau') }}                # [linux]
+    - {{ cdt('xorg-x11-util-macros') }}  # [linux]
+    - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
+    - {{ cdt('libx11-devel') }}          # [linux]
+    - {{ cdt('libx11-common') }}         # [linux]
+    - {{ cdt('libx11') }}                # [linux]
+    - {{ cdt('libxext-devel') }}         # [linux]
+    - {{ cdt('libxext') }}               # [linux]
+    - {{ cdt('libxrender-devel') }}      # [linux]
   host:
-    - glib
-    - gettext   # [unix and not (aarch64)]
-    - libiconv  # [win]
-    - libxml2
-    - cairo
-    - gdk-pixbuf
-    - pango
-  run:
-    - harfbuzz
-    - fontconfig
+    - gobject-introspection # [not win]
+    - glib >=2.50.0
+    - gettext   # [osx]
+    - harfbuzz >=2.0.0
+    - fontconfig  # [not win]
+    - libiconv  # [win and osx]
+    - libxml2 >=2.9.0
+    - cairo >=1.16.0
+    - gdk-pixbuf >=2.20
+    - pango >=1.46
+    - libpng
+    - libzip
     - freetype
+  run:
+    - harfbuzz >=2.0.0
+    - fontconfig  # [not win]
+    - libiconv  # [win or osx]
+    - libxml2 >=2.9.0
+    - cairo >=1.16.0
+    - gdk-pixbuf >=2.20
+    - pango >=1.46
```

Comment:
All these explicit constraints look dangerous. We should go with the constraints exported by run_exports as much as possible.

Reply:
I see what you mean. This leaves the max version open while run_exports sets the max bound. Using run_exports likely restricts the lower bound to something higher but that is an acceptable tradeoff (compared to actually search and set the appropriate upper bound on each dependency).